### ISSUE TITLE
3.8.0 run report has puzzling 'undefined' lines

### ISSUE THREAD

**Issue by PetMou** [cypress-io/cypress#5995]
When I do a 'cypress run' (windows 10 environment), I get several lines with text 'undefined' in the report.
To exclude any local influences, I made a dummy scenario which does absolutely nothing, with a minimal 'cypress.json' file. But still it reports these lines. 

```

  Running:  dummy.spec.js                                                                   (1 of 1)
undefined

  dummy describe
    √ dummy it (85ms)
undefined
undefined
  1 passing (96ms)
undefined

  (Results)

```
All tests pass, so it is **not** blocking. It's just puzzling.
Running in debug mode gives no clues as to _what_ is undefined:

```

 2019-12-18T09:00:40.155Z cypress:server:project onMocha pass
 2019-12-18T09:00:40.155Z cypress:server:reporter got mocha event 'pass' with args: [ { id: 'r3', title: 'dummy it', state: 'passed', body: 'function () {// nothing happens\n  }', type: 'test', duration: 15, wallClockStartedAt: '2019-12-18T09:00:40.145Z', timings: { lifecycle: 14, test: [Object] } } ]

     √ dummy it
 2019-12-18T09:00:40.155Z cypress:server:project onMocha test end
 2019-12-18T09:00:40.155Z cypress:server:reporter got mocha event 'test end' with args: [ { id: 'r3', title: 'dummy it', state: 'passed', body: 'function () {// nothing happens\n  }', type: 'test', duration: 15, wallClockStartedAt: '2019-12-18T09:00:40.145Z', timings: { lifecycle: 14, test: [Object] } } ]
 2019-12-18T09:00:40.169Z cypress:server:project onMocha test:after:run
 2019-12-18T09:00:40.169Z cypress:server:reporter got mocha event 'test:after:run' with args: [ { id: 'r3', title: 'dummy it', state: 'passed', body: 'function () {// nothing happens\n  }', type: 'test', duration: 15, wallClockStartedAt: '2019-12-18T09:00:40.145Z', wallClockDuration: 19, timings: { lifecycle: 14, test: [Object] } } ]
 2019-12-18T09:00:40.170Z cypress:server:project onMocha suite end
 2019-12-18T09:00:40.170Z cypress:server:reporter got mocha event 'suite end' with args: [ { id: 'r2', title: 'dummy describe', root: false, type: 'suite' } ]
undefined
 2019-12-18T09:00:40.171Z cypress:server:project onMocha suite end
 2019-12-18T09:00:40.171Z cypress:server:reporter got mocha event 'suite end' with args: [ { id: 'r1', title: '', root: true, type: 'suite' } ]
 2019-12-18T09:00:40.171Z cypress:server:project onMocha end
 2019-12-18T09:00:40.171Z cypress:server:reporter got mocha event 'end' with args: [ { end: '2019-12-18T09:00:40.167Z' } ]
 undefined
   1 passing (38ms)
 undefined
 
   (Results)
 
   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
   │ Tests:        1                                                                                │
   │ Passing:      1                                                                                │
   │ Failing:      0                                                                                │
   │ Pending:      0                                                                                │
   │ Skipped:      0                                                                                │
   │ Screenshots:  0                                                                                │
   │ Video:        true                                                                             │
   │ Duration:     0 seconds                                                                        │
   │ Spec Ran:     dummy.spec.js                                                                    │
   └────────────────────────────────────────────────────────────────────────────────────────────────┘

 2019-12-18T09:00:41.178Z cypress:server:run attempting to close the browser

>

**Comment by syedmuhdahmad** [comment]
@PetMou I am also  facing the same issue on Windows 10 any luck finding solution

**Comment by jennifer-shehane** [comment]
@PetMou Can you provide a small repo where when run where this is happening in Windows 10? Thanks.

**Comment by PetMou** [comment]
Hereby the minimal scenario that shows the issue. I hope this is enough?
[cypress_5995.zip](https://github.com/cypress-io/cypress/files/3997937/cypress_5995.zip)

These 'undefined' lines do not show up in our (Linux) pipeline, nor in a headed run.
We use Windows 10 Enterprise in a HyperV environment.
cypress.json:

```
{
  "chromeWebSecurity": false,
  "video": false,
  "integrationFolder": "cypress/integration"
}

```

**Comment by irfancharania** [comment]
I'm experiencing this same issue in our Windows Server 2016 Standard environment in a headless run with 3.8.0 and 3.8.1

**Comment by tmc101** [comment]
Seeing the same thing with 3.8.2 - this doesn't happen with 3.7.0.

To reproduce in an empty folder:

```
npm init -y

npm install cypress --save-dev

npx cypress open

npx cypress run
```

![image](https://user-images.githubusercontent.com/11647741/72258377-cf5df880-3605-11ea-9361-d58b884664ad.png)

**Comment by jennifer-shehane** [comment]
I was able to track this down in our own Appveyor runs on Windows of our kitchen-sink.

## 3.7.0

Appveyor build: https://ci.appveyor.com/project/cypress-io/cypress-example-kitchensink/builds/29572107

<img width="773" alt="Screen Shot 2020-01-14 at 1 59 50 PM" src="https://user-images.githubusercontent.com/1271364/72323100-4a8edf80-36d6-11ea-81e5-25b49bf8f547.png">

## 3.8.0

Appveyor build: https://ci.appveyor.com/project/cypress-io/cypress-example-kitchensink/builds/29572107

<img width="804" alt="Screen Shot 2020-01-14 at 2 00 01 PM" src="https://user-images.githubusercontent.com/1271364/72323109-4fec2a00-36d6-11ea-9ead-b81a9b7a4c8f.png">

**Comment by jennifer-shehane** [comment]
Is this reproducible when running `cypress run --browser chrome`?

**Comment by jennifer-shehane** [comment]
I've been looking through the PRs in 3.8.0, there are some that touch the `run.js` file that issues the output here, but I don't see anything that should have affected the output. 

- `--headless` chrome: https://github.com/cypress-io/cypress/pull/5676
- `--tag` flag: https://github.com/cypress-io/cypress/issues/832

There is also this Windows specific PR that went into 3.8.0 https://github.com/cypress-io/cypress/pull/5853

**Comment by dwelle** [comment]
> Is this reproducible when running `cypress run --browser chrome`?

Yes

**Comment by piksel** [comment]
Same behaviour on v4.0.1 

This is due to how `console.log` works in electron on windows: ([lib/browser/init.ts:18](https://github.com/electron/electron/blob/master/lib/browser/init.ts#L18))
```js
if (process.platform === 'win32') {
  // Redirect node's console to use our own implementations, since node can not
  // handle console output when running as GUI program.
  var consoleLog = function (...args) {
    return process.log(util.format(...args) + '\n')
  }
```
Which means that every time that mocha wants to output an empty row, through `console.log()`, this happens:
```js
let consoleLog = (format, ...args) => util.format(format, ...args); 
consoleLog(); // 'undefined'
```

**Comment by bahmutov** [comment]
nice @piksel - thanks for this. Is there an issue in Electron repo for this behavior by any chance? @flotwig can we stub `console.log()` and call `console.log('')` to get around this?

**Comment by piksel** [comment]
I haven't been able to find any issues for it in electron. I'll make one.

The temporary fix I did was to put this in `server/index.js`:
```js
const consoleLog = console.log;
console.log = (...args) => consoleLog(...(args.length<1?['']:args));
```

**Comment by PetMou** [comment]
I hope this doesn't alter the root cause found by @piksel, but the issue also appears when running with **headless chrome** options...

**Comment by piksel** [comment]
No, it still applies. Regardless of what browser you are using, the
"orchestrator" is always Cypress.exe (or the OS binary equivalent) which is
a branded electron binary.

**Comment by PetMou** [comment]
Fixed in 4.1.0 ! Closing the issue

**Comment by flotwig** [comment]
This was fixed because Electron was bumped from 7.1.10 to 7.1.13 in Cypress 4.1.0, and this bug was fixed in 7.1.13: https://github.com/electron/electron/pull/22173

**Comment by jennifer-shehane** [comment]
This issue will be closed to further comment as the exact issue here was resolved and tested. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
